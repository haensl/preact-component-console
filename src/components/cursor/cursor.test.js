import { h } from 'preact';
import render from 'preact-render-to-string';
import { shallow } from 'preact-render-spy';
import Cursor, { DEFAULTS } from './';

jest.useFakeTimers();

describe('Cursor', () => {
  describe('empty', () => {
    let tree;

    beforeEach(() => {
      tree = render(<Cursor />);
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}`));
    });

    it('does not add the default blink class', () => {
      expect(tree).not.toMatch(new RegExp(`${DEFAULTS.classes.blink}"`));
    });

    it('does not add the default write class', () => {
      expect(tree).not.toMatch(new RegExp(DEFAULTS.classes.write));
    });
  });

  describe('char', () => {
    let tree;

    beforeEach(() => {
      tree = render(<Cursor char="t"/>);
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}`));
    });

    it('adds the default write class', () => {
      expect(tree).toMatch(new RegExp(`${DEFAULTS.classes.write}"`));
    });

    it('does not add the default blink class', () => {
      expect(tree).not.toMatch(new RegExp(`${DEFAULTS.classes.blink}"`));
    });

    it('writes the char', () => {
      expect(tree).toMatch(/>t</);
    });
  });

  describe('classes', () => {
    describe('element', () => {
      let tree;

      beforeEach(() => {
        tree = render(<Cursor classes={{
          element: 'test-element'
        }} />);
      });

      it('renders as expected', () => {
        expect(tree).toMatchSnapshot();
      });

      it('adds the given class to the element', () => {
        expect(tree).toMatch(/class="test-element/);
      });
    });

    describe('blink', () => {
      let tree;
      let component;

      beforeEach(() => {
        tree = render(<Cursor classes={{
          blink: 'test-element--blink'
        }} />);
      });

      it('renders as expected', () => {
        expect(tree).toMatchSnapshot();
      });

      it('adds the default class to the element', () => {
        expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}`));
      });

      it('does not add the blink class, yet', () => {
        expect(tree).not.toMatch(new RegExp(`class="${DEFAULTS.classes.element} ${DEFAULTS.classes.blink}"`));
      });

      describe('after the default interval has passed', () => {
        beforeEach(() => {
          component = shallow(<Cursor classes={{
            blink: 'test-element--blink'
          }} />);
          jest.runTimersToTime(DEFAULTS.interval + 1);
          component.rerender();
          tree = component.toString();
        });

        it('adds the given blink class', () => {
          expect(tree).toMatch(/test-element--blink"/);
        });
      });
    });

    describe('write', () => {
      let tree;

      beforeEach(() => {
        tree = render(<Cursor char="t" classes={{
          write: 'test-element--write'
        }} />);
      });

      it('renders as expected', () => {
        expect(tree).toMatchSnapshot();
      });

      it('adds the default class to the element', () => {
        expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}`));
      });

      it('adds the given write class', () => {
        expect(tree).toMatch(/test-element--write"/);
      });
    });
  });
});
