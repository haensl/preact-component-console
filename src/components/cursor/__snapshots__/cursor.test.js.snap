// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cursor char renders as expected 1`] = `"<span class=\\"console-cursor console-cursor--write\\">t</span>"`;

exports[`Cursor classes blink after the default interval has passed after the default interval has passed another time renders as expected 1`] = `
"preact-render-spy (1 nodes)
-------
<span class=\\"console-cursor\\"></span>
"
`;

exports[`Cursor classes blink after the default interval has passed renders as expected 1`] = `
"preact-render-spy (1 nodes)
-------
<span class=\\"console-cursor test-element--blink\\"></span>
"
`;

exports[`Cursor classes blink renders as expected 1`] = `"<span class=\\"console-cursor\\"></span>"`;

exports[`Cursor classes element renders as expected 1`] = `"<span class=\\"test-element\\"></span>"`;

exports[`Cursor classes write renders as expected 1`] = `"<span class=\\"console-cursor test-element--write\\">t</span>"`;

exports[`Cursor empty renders as expected 1`] = `"<span class=\\"console-cursor\\"></span>"`;
